docs/DeveloperGuide.md

```diff
@@ -1,22 +1,222 @@
+making most attributes of all classes above private and avoiding getter methods if possible.
+The following diagram illustrates the resulting associations, methods and attributes.
+For the sake of clarity, UI class is omitted since it is used across most classes to systematize any form of
+printing output to the user.
+
+![High Level CD](images/highLevelClassDiagram.jpg)
+
+#### HealthMate
+Entry point to the application is the main function of HealthMate.
+The HealthMate class contains a private ChatParser attribute. This attribute's run function initiates, 
+after an initial greeting to the user, the interaction process. In this process, 
+the user enters commands with additional information into his command line application.
+The content of these commands is parsed by the ChatParser. 
+
+#### ChatParser
+The ChatParser class, instantiated once per application run, manages HealthMates overal usage 
+flow through its main run() method. 
+
+It has two primary attributes: 
+- A `MealEntriesList` object called `mealEntries` 
+  - Contains tracked calorie consumption 
+- A `MealList` object called `mealOptions` 
+  - Contains meals that are presaved by the user for quick selection to track commonly consumed meals
+
+These objects represent the application's underlying data with which the user interacts through the command line.
+To ensure no unintended changes are done, the ChatParser class orchestrates the effects of the users prompts.
+For saving these changes the ChatParser class makes use the `HistoryTracker`  which facilitates the process of 
+storing (and loading) User data, mealEntries data and mealOptions data to their corresponding files. 
+
+More details on the implementation of ChatParser follows in the Feature Section. 
+
+#### MealList
+The MealList class contains a private ArrayList of Meal object.
+Further, it encapsulates behaviour to operate on this list of meals. Most notably, 
+this adding or deleting a Meal to/from the list. 
+
+#### MealEntriesList
+The MealEntriesList class extends the MealList class. It overwrites the extractAndAppendMeal(...) method,
+and additionally includes methods specifically tailored to providing helpul user feedback, as the MealEntries stored 
+within its instance, signify the users calorie consumption. 
+As a MealEntry object differs from a Meal object by the additional timestamp attribute, this includes
+computations based on the time dimension. More specifically, the printDaysConsumptionBar() uses the UIs class'
+methods in the background to visualize the percentage of a certain days total consumption versus the idaeal consumption
+of a User class. 
+
+#### Meal
+The Meal class encapsulates the concept of a meal. As the purpose of this application 
+is to track calorie consumption, this consists of a mandatory calorie entry. The meal's name attribute, 
+is however an Optional<String> allowing a case, where no meaningful label can be attached to a certain consumption.
+![User SD](images/addMeal.png)
```

Comment:
![image](https://github.com/user-attachments/assets/89f6cdf8-f8f7-43c3-a74b-27381089b995)
Image fails to load